### ISSUE TITLE
tabulate fails on empty data with `headers` and `colalign`

### ISSUE THREAD

**Issue by Fifty-Nine** [astanin/python-tabulate#89]
It's possible to pass an empty `tabular_data` value to `tabulate` along with `headers` and this works as expected:

```
>>> import tabulate
>>> print(tabulate.tabulate([], headers=['foo', 'bar']))
foo    bar
-----  -----
>>> 
```

However, if `colalign` values are provided as well, it causes an exception:

```
>>> print(tabulate.tabulate([], headers=['foo', 'bar'], colalign=('left', 'right')))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/ssd/debian-next/venv/lib/python3.8/site-packages/tabulate.py", line 1494, in tabulate
    aligns[idx] = align
IndexError: list assignment index out of range
>>> 
```

My expectation is that this would behave as if `colalign` wasn't passed since there's no data to align.
I encountered this using tabulate 0.8.7.

**Comment by AlexanderStein** [comment]
I get the same backtrace, although my use case is slightly different. See below
```python
from tabulate import tabulate
table = [['', 2, 8, 25, 30], [5, 'x', 'x', 'x', 'x'], [23, 'x', 'x', 'x', '']]
#ok
print(tabulate(table, colalign="right"))
print(tabulate(table, colalign="left"))
print(tabulate(table))
#fail
print(tabulate(table, colalign="center"))
```
tabluate version 0.8.7

**Comment by astanin** [comment]
I can confirm this bug in 0.8.8.

**Comment by alexwlchan** [comment]
Digging into the issue a little, I think the problem in the second example is that `colalign` isn't meant to be a string?

The code is trying to set alignments `c`, `e`, `n`, and so on.

I think this is related to #84 – patch incoming shortly.

**Comment by eliegoudout** [comment]
@alexwlchan answered to @AlexanderStein already.
Regarding @Fifty-Nine's problem, it is fixed in #221 using directly`headersalign`:
```
>>> print(tabulate.tabulate([], headers=['foo', 'bar'], headersalign=('left', 'right')))
foo      bar
-----  -----
```

**Comment by Fifty-Nine** [comment]
> Regarding @Fifty-Nine's problem, it is fixed in #221 using directly`headersalign`:

Thanks for implementing this! I'll try it out when I get a chance.
